internal/app/machined/internal/phase/acpi/acpi.go

```diff
@@ -87,13 +96,42 @@ func listenForPowerButton() (err error) {
 				log.Printf("error reading from ACPI channel: %s", err)
 				return
 			}
+
 			if len(msgs) > 0 {
+				ok, err := parse(msgs, PowerButtonEvent)
+				if err != nil {
+					log.Printf("failed to parse netlink message: %v", err)
+					continue
+				}
+
+				if !ok {
+					continue
+				}
+
 				log.Printf("shutdown via ACPI received")
 				event.Bus().Notify(event.Event{Type: event.Shutdown})
+
 				return
 			}
 		}
 	}()
 
 	return nil
 }
+
+func parse(msgs []genetlink.Message, event string) (bool, error) {
+	for _, msg := range msgs {
+		ad, err := netlink.NewAttributeDecoder(msg.Data)
+		if err != nil {
+			return false, fmt.Errorf("failed to create attribute decoder: %w", err)
```

Comment:
should we continue here, as other messages won't be parsed and they might be the message we're looking for?

Reply:
good point ... probably need multi error too then.